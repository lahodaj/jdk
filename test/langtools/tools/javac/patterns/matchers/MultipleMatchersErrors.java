/**
 * @test
 * @enablePreview
 * @compile/fail/ref=MultipleMatchersErrors.out -XDrawDiagnostics MultipleMatchersErrors.java
 */

public class MultipleMatchersErrors {
    private static String test1(Object o) {
        if (o instanceof Person1(Object name, String username)) {
            return name + ":" + username;
        }
        return null;
    }

    private static String test2(Object o) {
        if (o instanceof Person1(Object name)) {
            return "" + name;
        }
        return null;
    }

    private static String test3(Object o) {
        if (o instanceof Person1(var name, String username)) {
            return name + ":" + username;
        }
        return null;
    }

    private static String test4(Object o) {
        if (o instanceof Person1(var name)) {
            return "" + name;
        }
        return null;
    }

    private static String test5(Object o) {
        if (o instanceof Person1(Class name)) {
            return "" + name;
        }
        return null;
    }

    public record Person1(String name, String username) {

        public Person1(String name) {
            this(name, name);
        }

        public __matcher Person1(String name, String username) {
             name = this.name;
             username = this.username;
        }

        public __matcher Person1(Integer id, String username) {
             id = 0;
             username = this.username;
        }

        public __matcher Person1(String name) {
             name = this.name;
        }

        public __matcher Person1(Integer id) {
             id = 0;
        }
    }

}
