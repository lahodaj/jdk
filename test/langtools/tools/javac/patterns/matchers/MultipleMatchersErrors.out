MultipleMatchersErrors.java:9:26: compiler.err.multiple.matchers: Person1(java.lang.Integer,java.lang.String),Person1(java.lang.String,java.lang.String)
MultipleMatchersErrors.java:16:26: compiler.err.multiple.matchers: Person1(java.lang.Integer),Person1(java.lang.String)
MultipleMatchersErrors.java:23:26: compiler.err.multiple.matchers: Person1(java.lang.Integer,java.lang.String),Person1(java.lang.String,java.lang.String)
MultipleMatchersErrors.java:30:26: compiler.err.multiple.matchers: Person1(java.lang.Integer),Person1(java.lang.String)
MultipleMatchersErrors.java:37:26: compiler.err.no.matchers
5 errors
